plugins/permission-node/src/integration/createConditionTransformer.ts

```diff
@@ -25,17 +25,31 @@ import {
   isOrCriteria,
 } from './util';
 
+const nonEmpty = <T>(list: T[]): [T, ...T[]] => {
+  if (list.length === 0) {
+    throw new Error(
+      'Invalid conditions provided. Expected at least one item for `allOf`/`anyOf` criteria but received none.',
+    );
+  }
+
+  return list as [T, ...T[]];
+};
```

Comment:
Can this be expressed as a type assertion?

```suggestion
const nonEmpty = <T>(list: T[]): list is [T, ...T[]] => {
  if (list.length === 0) {
    throw new Error(
      'Invalid conditions provided. Expected at least one item for `allOf`/`anyOf` criteria but received none.',
    );
  }
};
```

Reply:
Ah I see! That totally makes sense. It does leave me a bit confused about something tangentially related though - given that this code is validating part of an object of type `PermissionCriteria`, shouldn't the type already guarantee that the array won't be empty?

Reply:
Yeah that's what I had hoped, but calling `map` on a `NonEmptyArray` doesn't preserve it's "non-emptiness"

Reply:
Yup totally agree — it shouldn't be possible to reach that `throw` so casting feels more appropriate here anyway